db/migrate/20170217194851_create_firmwares.rb

```diff
@@ -0,0 +1,12 @@
+class CreateFirmwares < ActiveRecord::Migration[5.0]
+  def change
+    create_table :firmwares do |t|
+      t.string   :name
+      t.string   :build
+      t.string   :version
+      t.string   :ph_server_uuid, :index => true
```

Comment:
This relation should probably hang off of `hardwares`.  It could be used by other provider types too.